java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4586,6 +4586,44 @@ public void srandmember(BaseClient client) {
         assertInstanceOf(RequestException.class, executionExceptionWithCount.getCause());
     }
 
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void spop_spopCount(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        String stringKey = UUID.randomUUID().toString();
+        String nonExistingKey = UUID.randomUUID().toString();
+        String member1 = UUID.randomUUID().toString();
+        String member2 = UUID.randomUUID().toString();
+        String member3 = UUID.randomUUID().toString();
+
+        assertEquals(1, client.sadd(key, new String[] {member1}).get());
+        assertEquals(member1, client.spop(key).get());
+
+        assertEquals(3, client.sadd(key, new String[] {member1, member2, member3}).get());
+        assertEquals(Set.of(member1, member2, member3), client.spopCount(key, 4).get());
+        assertEquals(0, client.scard(key).get());
+
+        assertEquals(3, client.sadd(key, new String[] {member1, member2, member3}).get());
+        assertEquals(Set.of(), client.spopCount(key, 0).get());
+
+        assertNull(client.spop(nonExistingKey).get());
+        assertEquals(Set.of(), client.spopCount(nonExistingKey, 3).get());
+
+        // invalid argument - count must be positive
```

Comment:
Please add (or modify) a test where count > set size

Reply:
already have that test on line 4604, I'll add a comment to it though